sequencer/src/block/payload.rs

```diff
@@ -155,13 +152,19 @@ impl<TableWord: TableWordTraits> Payload<TableWord> {
 
     pub fn from_txs(
         txs: impl IntoIterator<Item = <payload::Payload<TxTableEntryWord> as BlockPayload>::Transaction>,
+        chain_config: &ChainConfig,
     ) -> Result<Self, Error> {
         let mut namespaces: HashMap<NamespaceId, NamespaceInfo> = Default::default();
         let mut structured_payload = Self {
             raw_payload: vec![],
             ns_table: NameSpaceTable::default(),
         };
+        let mut block_size = 0u64;
         for tx in txs.into_iter() {
+            block_size += tx.payload().len() as u64;
```

Comment:
This is not quite right as the payload contains other metadata, like a transaction table for each namespace, that could cause it to exceed the max block size even if the sum of transaction payloads doesn't.

Not sure of the best solution, maybe @ggutoski has some ideas. One idea could be to add a `len()` function to `NamespaceInfo`, which would return something like `TxTableEntry::byte_len() + tx_table.len() + tx_bodies.len()` (the first term is for the one word used to encode the length of the tx table). Then we would basically loop until `namespaces.values().map(NamespaceInfo::len).sum() > chain_config.max_block_size()`.

Reply:
I'm currently rewriting everything in `block`. I can definitely accommodate this. For this PR I suggest you do as little as possible to get it working.

It should be possible to keep a running count of the block byte length for each new transaction *that includes tx table sizes*, then just stop when we hit the limit.

(Note that a single large tx could cause a lot of wasted block space if it happens to be the one that triggers `break`. We might need to do something more intelligent in the future.)

Happy to hop on a call to nail this down.

Reply:
Now that I think of it, maybe this is easy. Just do
```rust
block_size += tx.payload.len() + [size of a tx table entry];
```

Reply:
that's a tiny amount of data, but it could in principle push you over the limit. You could account for it on-the-fly by doing `block_size += size_of::<TxTableEntry>()` each time you encounter a *new* namespace. Unfortunately that's wrapped up in `update_namespace_with_tx`, so you might need to hack that method to get what you want. I suggest you don't worry about ugly code for this PR though because it will all be replaced soon.

Reply:
I'm still going over the tests to make sure they are still valid, but I *think* 61dd04f5b9a2db478af802ac29ac8344e8824ac6 gets us what we need. Note that I removed a `TODO` that seemed relevant.

Reply:
i like it!